### PR TITLE
SPR1-1737: New mvf_copy.py script

### PR DESCRIPTION
**Pull request by ludwigschwardt**
This makes a local copy of an MVF4 dataset (local or remote). In the
process it also allows stripping out correlation products. It currently
only supports extracting the autocorrelations via the `--corrprods=auto`
script option. The filtered RDB file and chunks are stored in a new
subdirectory (by default "./archive").

The script has much in common with `mvf_rechunk.py` and might eventually
merge with it.

### CODE REVIEW

File: scripts/mvf_copy.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    kwargs['corrprods'] = args.corrprods
+d.select(**kwargs)
+
+# Convenience variables
+store = d.source.data.store
+cbid = d.source.capture_block_id
+stream = d.source.stream_name
+telstate = d.source.telstate
+corrprod_mask = d._corrprod_keep
+rdb_filename = Path(urlparse(d.url).path).name
```
Comment (Jordatious) [inline scripts/mvf_copy.py:78]: I think this should be `args.url`

**Review by Jordatious — APPROVED** [review]
Looks good! With that small correction of `d.url` to `args.url`, this works for both local MVF data on IDIA disk, and remote MVF data on the SARAO archive. I did encounter several read timeouts when pulling remote MVF data (~50 GB). I'm not sure if anything can easily be done about that.

File: scripts/mvf_copy.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+parser = argparse.ArgumentParser()
+parser.add_argument('url', help='Dataset URL (or input RDB file path)')
+parser.add_argument('-o', '--output', type=Path,
+                    default=Path.cwd() / 'archive',
+                    help='Output directory (defaults to "./archive")')
+parser.add_argument('--corrprods',
+                    help='Select correlation products (kwarg to katdal.DataSet.select). '
+                         'Keeps all corrprods by default.')
+parser.add_argument('--workers', type=int, default=8 * dask.system.CPU_COUNT,
```
[the remark this answers is not in the document]
Comment (ludwigschwardt) [reply scripts/mvf_copy.py]: For the record, `dask.system.CPU_COUNT` was added in dask 2.7.0 (katdal still currently expects dask >= 1.2.1).

File: scripts/mvf_copy.py [hunk 3]
```diff
@@ -23,103 +23,116 @@
...
+        kwargs['corrprods'] = args.corrprods
+    d.select(**kwargs)
+
+    # Convenience variables
+    cbid = d.source.capture_block_id
+    stream = d.source.stream_name
+    telstate = d.source.telstate
+    # XXX Replace private member with public corrprod index member when it exists
+    corrprod_mask = d._corrprod_keep
+    rdb_filename = Path(urlparse(args.source).path).name
```
[the remark this answers is not in the document]
Comment (ludwigschwardt) [reply scripts/mvf_copy.py]: Done.

File: scripts/mvf_copy.py [hunk 4]
```diff
@@ -23,103 +23,116 @@
...
+
+    telstate_overrides = katsdptelstate.TelescopeState()
+    # Override bls_ordering in telstate (in stream namespace) to match dataset selection
+    telstate_overrides.view(stream)['bls_ordering'] = d.corr_products
```
[the remark this answers is not in the document]
Comment (ludwigschwardt) [reply scripts/mvf_copy.py:113]: Updated (even though katdal doesn't need it).

File: scripts/mvf_copy.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ludwigschwardt) [reply scripts/mvf_copy.py]: Bumped dask requirement to 2.7.0.

**Pull request merged**